Automattic/wp-calypso
# Split out `mc` from `lib/analytics`

This is the first of several PRs that will make `lib/analytics` more modular, breaking up its monolithic default export. This will eventually result in being able to contain larger parts of analytics to just the sections they're needed in, instead of loading everything upfront regardless of the current route.

The PR moves `mc` to its own module under `lib/analytics` and updates all code accordingly.

Note that while I fixed a few easy lint issues, I felt that others were out of scope for this PR, and I let them be. This means that the lint test will fail due to the pre-existing issues.

#### Changes proposed in this Pull Request

* Extract `mc` into its own module
* Update all code accordingly
* Minor lint fixes

#### Testing instructions

There are no code changes, so it should be sufficient to ensure that there are no build errors or unit test failures related to e.g. an import not being correctly updated.
